File: sumRange/sumRange.rb
def sum(array)
  sum = 0
  array.length.times |i|
    sum += array[i]
  end
  sum
end

def range(x, y, z=1)
  array = []
  i = x
  if z == 0 || x == y
    array = [i]
  elsif y > x
    while i <= y && i >= x
      array << i
      i = i + z
    end
  else
    while i >= y && i <= x
      array << i 
      i = i + z
    end
  end
  array
end

File: sumRange/sumRange.js
function sum(array) {
  let total = 0
  for ( i = 0 ; i < array.length ; i++ ) {
    total += array[i]
  }
  return total
}

function range(x, y, z=1) {
  let array = []
  let i = x
  if (z === 0 || x === y) {
    array = [i]
  } else if (y > x) {
    while (i <= y && i >= x) {
      array.push(i)
      i += z
    }
  } else {
    while (i >= y && i <= x) {
      array.push(i)
      i += z
    }
  }
  return array
}